docs/proposal-0004-jetstream-preview.md

```diff
@@ -0,0 +1,118 @@
+
+The queries that get executed will be saved locally into files which makes it possible for users to debug the queries (enrollment and metrics) that get executed.
+
+Running the preview command will also print out cost estimates for running analyses on the experiment.
+
+The jetstream CI will need to be updated to publish jetstream to pypi. This way users do not need to pull in the jetstream repository, but can simply run `pip install mozilla-jetstream`.
+
+Jetstream fetches information on how experiments are configured from the Experimenter API. In some cases users might want to get a preview for experiments that haven't been launched yet or that haven't seen any enrollments. For these experiments a random population sample can be generated when specifying `--generate-population` that will be considered as the clients that enrolled.
+
+### Other ideas
+
+Metadata around cost could be stored in a temporary BigQuery table and displayed on the preview dashboard. It might be possible to also store and display the queries that were run as part of the preview including errors that occured. This would help with debugging queries early on.
+
+## Experimenter API
+
+Experimenter only has an API that surfaces experiments that are currently live. To run the preview on experiments that are currently in draft mode, changes would need to be made to surface these experiments through the API.
+
+Having access to experiments in preview would also help users to write and validate configs before the experiment has launched. Currently any pull-request adding a config for an experiment that isn't live cannot be merged or even be validated.
+
+## Risks
+
+There are some potential risks with providing users with the option of generating previews:
+* Users might optimize their configs based on the preview data which could impact the experiment results
+    * This is probably the biggest risk, however users might already do this, but instead rerun the entire experiment multiple times until they see the results they'd like.
+* Creating a preview might take too long
+    * Depending on the experiment size, and the provided preview parameters, generating a preview can take a while until results are available. Or in the worst case, if too much data is being processed the preview might not even finish when run locally.
+* With preview results being displayed in Looker instead of Experimenter it could be confusing for users to interpret the visualizations
+    * The Looker preview dashboard will show a subset of the results and without the explanations that are part of the Experimenter interface. Users that are less familiar with interpreting experiment data might be confused by that.
```

Comment:
This is interesting. Would it be useful to display results in Experimenter UI? I'm already thinking that the Results page should maybe be called Preview Results until the experiment is complete to help head off some of the confusion we already get.

Reply:
Yeah, that's definitely an interesting idea. I guess we'd need to communicate this to users well, since the results could suddenly change once the "real" results become available.